common/prometheus-monitors/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,49 @@
+{{- if .Values.serviceMonitor.enabled }}
+{{- if .Values.podMonitor.enabled }}
+{{ fail "failed to render ServiceMonitor. PodMonitor is also enabed. Only one is allowed."}}
+{{- end }}
```

Comment:
{{- if and .Values.podMonitor.enabled .Values.serviceMonitor.enabled }}
{{ fail "Could not render Servicemonitor. '.Values.podMonitor.enabled' and '.Values.serviceMonitor.enabled' are mutually exclusive."}}
{{- end }}

Reply:
This is intentionally split. The outer if is for determining if the template should be rendered the inner if to fail if both service and pod monitor are enabled.

Reply:
All good.